### PR TITLE
6 get all comments by article

### PR DESCRIPTION
**Pull request by NaveedGujral**

### CODE REVIEW

File: __tests__/integration-tests.test.js [hunk 1]
```diff
@@ -14,25 +14,68 @@ afterAll(() => {
...
     })
 });
 
+describe("GET /api/articles/:article_id/comments tests", () => {
+    test('responds with array of objects with the correct keys. The array has the correct length and the accompanying code should be status 200', () => {
+        return request(app)
+        .get("/api/articles/3/comments")
+        .expect(200)
+        .then(( { body }) => {
+            console.log(body.comments)
```
Comment (aloush-dev) [inline __tests__/integration-tests.test.js]: remember to remove console logs

File: models/models.js [hunk 2]
```diff
@@ -8,6 +8,21 @@ exports.selectAllTopics = () => {
...
+    .then(({ rows }) => {
+        const commentArray = rows
+
+        if (commentArray.length === 0) {
+            return Promise.reject({
+                status: 404,
+                msg: `this article has no comments`
+                })
+            }
+        return commentArray
```
Comment (aloush-dev) [inline models/models.js]: This isn't an error we should be returning the empty array back to the user.

**Review by aloush-dev — CHANGES_REQUESTED** [review]
I'd like to see a test for a 200 success when the article ID is valid but there is no comments.

Please add the missing test and change the functionality then resubmit.

Thanks

**Review by aloush-dev — CHANGES_REQUESTED** [review]
I'd like to see a test for a 200 success when the article ID is valid but there is no comments.

Please add the missing test and change the functionality then resubmit.

Thanks

File: errors.js [hunk 3]
```diff
@@ -1,24 +1,27 @@
...
-    } 
-  };
 
 exports.serverErrors = (err, req, res, next) => {
-    console.log("server error: ", err)
-    if (err.status) {
-      res.status(500).send({ msg: "internal server error" });
-    } 
-  };
+  console.log("server error: ", err)
```
Comment (MKD01) [inline errors.js]: and this one

File: errors.js [hunk 4]
```diff
@@ -1,24 +1,27 @@
+exports.customErrors = (err, req, res, next) => {
+  if (err.status) {
+    res.status(err.status).send({ msg: err.msg });
+  }
+  else next(err) 
+};
+
 exports.psqlErrors = (err, req, res, next) => {
-    console.log("PSQL error code: ",err.code)
+  console.log("PSQL error code: ",err.code)
```
Comment (MKD01) [inline errors.js]: Dont forget to remove console.logs

File: endpoints.json [hunk 5]
```diff
@@ -25,5 +25,29 @@
         }
       ]
     }
+  },
+  "GET /api/articles/:article_id/comments": {
+    "description": "responds with an array of comments that match the article id",
+    "queries": ["article_id"],
+    "exampleResponse": {
```
Comment (MKD01) [inline endpoints.json]: article_id is not a query here but a parametric endpoint which you have already showed with the : symbol, and this endpoint does not have any queries yet so this should be an empty array

**Review by MKD01 — APPROVED** [review]
Great work adding in the requested changes, I have added some comments for you to take a look at, and once you are happy feel free to merge this in.

**Pull request merged**